src/Messages/TSMessagesManager.m

```diff
@@ -914,12 +916,12 @@ - (TSIncomingMessage *_Nullable)handleReceivedEnvelope:(OWSSignalServiceProtosEn
                                                              messageBody:body
                                                            attachmentIds:attachmentIds
                                                         expiresInSeconds:dataMessage.expireTimer];
+          DDLogDebug(@"%@ incoming 1:1 text message: %@", self.tag, incomingMessage.debugDescription);
+          [incomingMessage saveWithTransaction:transaction];
```

Comment:
We were saving messages twice in some cases.  See comments above and below.  We should either save once when the message is created (as I've done here) or once later.  I figured the latter was a little easier to ensure to be correct.